app/models/user.rb

```diff
@@ -1,7 +1,10 @@
 # app/models/user.rb
 class User < ApplicationRecord
     validates :email, presence: true, uniqueness: true
+    validates :email, format: { with: URI::MailTo::EMAIL_REGEXP, message: 'must be a valid email address' }
 
-    # Relationship with LmsCredential
+    # Associasions
     has_many :lms_credentials, dependent: :destroy
+    has_many :user_to_courses
```

Comment:
this is a strange variable/class name, and isn't plural (which i'd expect for has_many).  do you mean something like `course_specific_users`? it's worth taking time to think carefully about names, as once they are committed you're generally stuck with them